src/main/java/org/opensearch/security/action/configupdate/TransportConfigUpdateAction.java

```diff
@@ -125,8 +125,10 @@ protected ConfigUpdateResponse newResponse(
 
     @Override
     protected ConfigUpdateNodeResponse nodeOperation(final NodeConfigUpdateRequest request) {
-        configurationRepository.reloadConfiguration(CType.fromStringValues((request.request.getConfigTypes())));
-        backendRegistry.get().invalidateCache();
+        if (dynamicConfigFactory.isBackgroundThreadComplete()) {
+            configurationRepository.reloadConfiguration(CType.fromStringValues((request.request.getConfigTypes())));
```

Comment:
`resolveConfiguration` is used in 2 places. Here and within the bgThread. If I bake in a check like this into `configurationRepository.reloadConfiguration` then it would need to know which context it was run in because if its within the bgThread then we do want it to reload the configuration. I think it makes sense to put this check here before the reloadConfiguration is called.

Reply:
Gotcha; however, I'm saying this change in behavior requires the caller to understand internal behavior of the reload configuration logic, "there is another property on a config repo that knows if the bg thread has completed or not, callers shouldn't attempt to reload the configuration unless this has completed"

By pushing this information / check outside of the config repository it creates coupling between these classes this coupling is confusing.

I'm suggesting that only the config repository should need to know this information, that it can track if it has completed loading the configuration and thus is in the best position to know how to process a reload method call.  Consider the following interface:

```
public interface ConfigRepo {
   /** Called when the configuration repo should initialize itself across the cluster*/
   initialize();

   /** Synchronizes the configuration across all nodes in the cluster, returns the synchronization result */
   ConfigStatus synchronizeConfig();
}

public enum ConfigSyncResult {
   ALREADY_UP_TO_DATE,
   INITALIZED,
   RELOADED_FROM_CLUSTER;
}
```

With this kind of interface the following would be a clearer intention `if (configRepo.synchronizeConfig() == RELOADED_FROM_CLUSTER) { backendRegistry.invalidate() }`

Reply:
@peternied I updated this PR to remove the check here now. It now uses a threadContext transient header to differentiate where `reloadConfiguration` is called so that it permits the request to proceed if its within the bgThread.